Fruitties/iosApp/iosApp/ContentView.swift

```diff
@@ -67,37 +67,37 @@ struct FruittieView: View {
 }
 
 class UIModel: ObservableObject {
-    let dataRepository : DataRepository
-    init(dataRepository: DataRepository) {
-        self.dataRepository = dataRepository
+    let mainViewModel: MainViewModel
+    init(mainViewModel: MainViewModel) {
+        self.mainViewModel = mainViewModel
     }
     @Published
     private(set) var fruitties: [Fruittie] = []
     @Published
-    private(set) var cart: Cart = Cart(items: [])
+    private(set) var cartDetails: [CartItemDetails] = []
 
     @MainActor
-    func observeDatabase() async {
-        for await fruitties in dataRepository.getData() {
-            self.fruitties = fruitties
+    func observeHomeUiState() async {
+        for await homeUiState in mainViewModel.homeUiState {
+            self.fruitties = homeUiState.fruitties
         }
     }
     
     @MainActor
-    func watchCart() async {
-        for await cart in dataRepository.getCart() {
-            self.cart = cart
+    func observeCartUiState() async {
+        for await cartUiState in mainViewModel.cartUiState {
+            self.cartDetails = cartUiState.cartDetails
         }
     }
     
     func addToCart(fruittie: Fruittie) async {
```

Comment:
Here and below - does this still need be async?

Reply:
I think so because cartUiState is a Flow<> and this for loop spins forever collecting data, but maybe there is another way?

If I remove `async`, I get an error for `await`.
`'async' in a function that does not support concurrency`

If I remove `await`, I get an error with `mainViewModel.cartUiState` in the for-loop.
`For-in loop requires 'SkieSwiftStateFlow<CartUiState>' to conform to 'Sequence'`

Reply:
Oh I did not realize we were using SKIE to generate proper coroutine apis for Swift.

Although, I am actually talking about this function `addToCart`, which doesn't look like it uses `await` anymore?